apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/RpcRetryConsumer.cs

```diff
@@ -174,38 +174,31 @@ private void FlushBuffer()
                 }
                 toSend = _buffer;
                 toSendSize = _sizeInBytes;
+                _sizeInBytes = 0;
                 _buffer = new List<T>();
                 _retries++;
             }
 
             try
             {
                 _consumer.Receive(toSend);
-
-                // The call succeeded, remove update the buffer size and retries.  
                 lock (_mutex)
                 {
-                    _sizeInBytes -= toSendSize;
                     _retries = 0;
                 }
             }
-            catch (RpcException ex) 
+            catch (RpcException) 
             {
-                // If we have reached the maximum number of retries. 
                 if (_retries >= _options.RetryAttempts)
                 {
-                    if (_options.ExceptionHandling == ExceptionHandling.Propagate)
+                    lock (_mutex)
                     {
-                        throw ex;
+                        _retries = 0;
```

Comment:
Nope, I'm wrong about that. But you can remove the lock if you change the increment of `_retries` to use `Interlocked.Increment` instead.